akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -753,7 +754,30 @@ trait DefaultSSLContextCreation {
   protected def system: ActorSystem
   protected def sslConfig: AkkaSSLConfig
 
-  protected def createDefaultClientHttpsContext(): HttpsConnectionContext = {
+  // --- log warnings ---
+  private[this] def log = system.log
+
+  def validateAndWarnAboutLooseSettings() = {
+    if (sslConfig.config.loose.disableHostnameVerification)
+      log.warning("Detected that Hostname Verification (via ssl-config) is disabled globally for the Http extension! " +
+        "This is very dangerous and may expose you to man-in-the-middle attacks. " +
+        "If you are forced to interact with a server that is behaving such that you must disable this setting, " +
+        "please disable it for a given connection instead, by configuring a specific HttpsConnectionContext " +
+        "for use only for the trusted target that hostname verification would have blocked.")
+
+    if (sslConfig.config.loose.disableSNI)
+      log.warning("Detected that Server Name Indication (SNI) is disabled globally (via ssl-config) for the Http extension! " +
+        "This is very dangerous and may expose you to man-in-the-middle attacks. " +
+        "If you are forced to interact with a server that is behaving such that you must disable this setting, " +
+        "please disable it for a given connection instead, by configuring a specific HttpsConnectionContext " +
+        "for use only for the trusted target that hostname verification would have blocked.")
```

Comment:
Refer to `sslConfig.config.loose.disableSNI` here? Looks like there`s some DRY potential here. ;)